lib/mongo/collection/view/writable.rb

```diff
@@ -538,6 +541,44 @@ def update_one(spec, opts = {})
             end
           end
         end
+
+        private
+
+        # Checks the update documents to make sure they only have atomic modifiers.
+        # Note that as per the spec, we only have to examine the first element
+        # in the update document.
+        #
+        # @param [ Hash | Array<Hash> ] spec The update document or pipeline.
+        #
+        # @raise [ Error::InvalidUpdateDocument ] if the first key in the
+        #   document does not start with a $.
+        def validate_update_documents!(spec)
+          if update = spec.is_a?(Array) ? spec&.first : spec
```

Comment:
Here we could also use `if update = Array(spec).first`

Reply:
see my comment above